### PR TITLE
Simulation time bugfix

### PR DESCRIPTION
**Pull request by mkvanhooft**
the stack run method did not have a ns.sim_reset() anywhere so multiple calls would not reset simulation time
Generic qdevice was missing S and T gates

### CODE REVIEW

**Comment by dieriver** [comment]
In general, I don't an issue with this PR.
I checked the NetQASM operation, and it seems they only support  S and T gates for NV qprocs. Moreover, in this case, the S and T instructions are transpiled into the equivalent rotations. Assuming that the "Vanilla" flavor of NetQASM maps to the "generic qproc" in squidasm, the vanilla flavor of qprocs do not support S and T gates so far, and it seems there are not NetQASM tests covering those scenarios.
Bottom line: I think this is safe to merge, but I think we should discuss a bit further with Bart once he is back.

**Review by dieriver — APPROVED** [review]

File: tests/stack/test_stack_run.py [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+            epr_sockets=[self.peer_name],
+            max_qubits=2,
+        )
+
+    def run(self, context: ProgramContext):
+        yield from teleport_recv(context, peer_name=self.peer_name)
+        self.complete = True
+
+
+class TestRun(unittest.TestCase):
```
Comment (dieriver) [inline tests/stack/test_stack_run.py:53]: I don't see how this test case is relevant for the S and T gates.... maybe it  tests the bugfix also in this PR?

Comment (mkvanhooft) [reply tests/stack/test_stack_run.py:53]: This is used to check the `run` method. This is a testcase covering the simulation time issue

**Review by dieriver — APPROVED** [review]

**Review by maxsnoodijk — APPROVED** [review]

**Review by dieriver — APPROVED** [review]
Approved
